model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py

```diff
@@ -1332,6 +1744,54 @@ async def _create_or_update_resources(
                     destination_rule=destination_rule_template,
                     name=k8s_resource_group_name,
                 )
+        elif (
+            model_endpoint_record.endpoint_type
+            in {
+                ModelEndpointType.SYNC,
+                ModelEndpointType.STREAMING,
+            }
+            and request.build_endpoint_request.nodes_per_worker > 1
```

Comment:
Any way to clean/organize these cases? Does it make sense to nest the is_multinode check in the endpoint type check instead of making it flat?

Reply:
I feel like it's probably nicer to keep things flat, since there are some commonly created resources (and it's nice to group things by function)